===================================================================
RCS file: /cvs/gnome/libgtop/sysdeps/linux/siglist.c,v
retrieving revision 1.7
retrieving revision 1.8
diff -u -r1.7 -r1.8
--- siglist.c	2005/03/25 12:52:12	1.7
+++ siglist.c	2005/05/25 22:28:42	1.8
@@ -35,40 +35,110 @@
 
 const glibtop_signame glibtop_sys_siglist [] =
 {
+#ifdef SIGHUP
 	{SIGHUP,	"SIGHUP",       "Hangup (POSIX)."},
+#endif
+#ifdef SIGINT
 	{SIGINT,	"SIGINT",       "Interrupt (ANSI)."},
+#endif
+#ifdef SIGQUIT
 	{SIGQUIT,       "SIGQUIT",      "Quit (POSIX)."},
+#endif
+#ifdef SIGILL
 	{SIGILL,	"SIGILL",       "Illegal instruction (ANSI)."},
+#endif
+#ifdef SIGTRAP
 	{SIGTRAP,       "SIGTRAP",      "Trace trap (POSIX)."},
+#endif
+#ifdef SIGABRT
 	{SIGABRT,       "SIGABRT",      "Abort (ANSI)."},
+#endif
+#ifdef SIGIOT
 	{SIGIOT,	"SIGIOT",       "IOT trap (4.2 BSD)."},
+#endif
+#ifdef SIGBUS
 	{SIGBUS,	"SIGBUS",       "BUS error (4.2 BSD)."},
+#endif
+#ifdef SIGFPE
 	{SIGFPE,	"SIGFPE",       "Floating-point exception (ANSI)."},
+#endif
+#ifdef SIGKILL
 	{SIGKILL,       "SIGKILL",      "Kill, unblockable (POSIX)."},
+#endif
+#ifdef SIGUSR1
 	{SIGUSR1,	"SIGUSR1",      "User-defined signal 1 (POSIX)."},
+#endif
+#ifdef SIGSEGV
 	{SIGSEGV,	"SIGSEGV",      "Segmentation violation (ANSI)."},
+#endif
+#ifdef SIGUSR2
 	{SIGUSR2,	"SIGUSR2",      "User-defined signal 2 (POSIX)."},
+#endif
+#ifdef SIGPIPE
 	{SIGPIPE,       "SIGPIPE",      "Broken pipe (POSIX)."},
+#endif
+#ifdef SIGALRM
 	{SIGALRM,       "SIGALRM",      "Alarm clock (POSIX)."},
+#endif
+#ifdef SIGTERM
 	{SIGTERM,       "SIGTERM",      "Termination (ANSI)."},
+#endif
+#ifdef SIGSTKFLT
 	{SIGSTKFLT,     "SIGSTKFLT",    "Stack fault."},
+#endif
+#ifdef SIGCLD
 	{SIGCLD,	"SIGCLD",       "Same as SIGCHLD (System V)."},
+#endif
+#ifdef SIGCHLD
 	{SIGCHLD,       "SIGCHLD",      "Child status has changed (POSIX)."},
+#endif
+#ifdef SIGCONT
 	{SIGCONT,       "SIGCONT",      "Continue (POSIX)."},
+#endif
+#ifdef SIGSTOP
 	{SIGSTOP,       "SIGSTOP",      "Stop, unblockable (POSIX)."},
+#endif
+#ifdef SIGTSTP
 	{SIGTSTP,       "SIGTSTP",      "Keyboard stop (POSIX)."},
+#endif
+#ifdef SIGTTIN
 	{SIGTTIN,       "SIGTTIN",      "Background read from tty (POSIX)."},
+#endif
+#ifdef SIGTTOU
 	{SIGTTOU,       "SIGTTOU",      "Background write to tty (POSIX)."},
+#endif
+#ifdef SIGURG
 	{SIGURG,	"SIGURG",       "Urgent condition on socket (4.2 BSD)."},
+#endif
+#ifdef SIGXCPU
 	{SIGXCPU,       "SIGXCPU",      "CPU limit exceeded (4.2 BSD)."},
+#endif
+#ifdef SIGXFSZ
 	{SIGXFSZ,       "SIGXFSZ",      "File size limit exceeded (4.2 BSD)."},
+#endif
+#ifdef SIGVTALRM
 	{SIGVTALRM,     "SIGVTALRM",    "Virtual alarm clock (4.2 BSD)."},
+#endif
+#ifdef SIGPROF
 	{SIGPROF,       "SIGPROF",      "Profiling alarm clock (4.2 BSD)."},
+#endif
+#ifdef SIGWINCH
 	{SIGWINCH,      "SIGWINCH",     "Window size change (4.3 BSD, Sun)."},
+#endif
+#ifdef SIGPOLL
 	{SIGPOLL,       "SIGPOLL",      "Pollable event occurred (System V)."},
+#endif
+#ifdef SIGIO
 	{SIGIO,		"SIGIO",	"I/O now possible (4.2 BSD)."},
+#endif
+#ifdef SIGPWR
 	{SIGPWR,	"SIGPWR",       "Power failure restart (System V)."},
+#endif
+#ifdef SIGSYS
 	{SIGSYS,	"SIGSYS",       "Bad system call."},
+#endif
+#ifdef SIGUNUSED
 	{SIGUNUSED,	"SIGUNUSED",	""},
+#endif
 	{0,		NULL,		NULL}
 };
